src/validation.cpp

```diff
@@ -3832,6 +3832,18 @@ bool ChainstateManager::AcceptBlockHeader(const CBlockHeader& block, BlockValida
     if (ppindex)
         *ppindex = pindex;
 
+    const auto msg = strprintf("saw new valid header hash=%s", hash.ToString());
+
+    if (!ActiveChainstate().IsInitialBlockDownload()) {
+        // These messages are valuable for detecting potential
+        // selfish mining behavior; if multiple displacing headers are seen
+        // near simultaneously across many nodes in the network, this might be
+        // an indication of selfish mining.
+        LogPrintf("%s\n", msg);
```

Comment:
Could just make this:

```c++
const auto level{(IsIBD() ? BCLog::Level::Info : BCLog::Level::Warning)};
LogPrintf(BCLog::VALIDATION, level, "saw new ...", hash.ToString());
```

(or `None` rather than `Warning` maybe)

Reply:
Ah, TIL:

https://github.com/bitcoin/bitcoin/blob/40e1c4d4024b8ad35f2511b2e10bf80c5531dfde/src/logging.cpp#L127-L131